### PR TITLE
feat: add RemoveBackgroundAndResizeNode — standalone PIL-only backgro…

### PR DESCRIPTION
**Pull request by Limbicnation**
…und removal with resize

- Single-file node combining background removal + resize (no external background_remover.py dependency)
- PIL-only implementation: k-means colour clustering, dominant-bg estimation from edges, alpha binarisation
- ScalingMixin from grabcut_nodes.py: power-of-8 resize, 6 preset sizes + ORIGINAL
- ComfyUI contract: INPUT_TYPES, RETURN_TYPES (IMAGE + MASK), FUNCTION, CATEGORY
- Registered in __init__.py alongside existing TransparencyBackgroundRemover and GrabCut nodes
- structlog logging, torch.no_grad() guard, full validation

## Summary by CodeRabbit

* **New Features**
  * Added "Remove Background & Resize" node for ComfyUI, combining background removal and image resizing in a single operation with configurable parameters for tolerance, color clustering, output size, and scaling interpolation methods.

### CODE REVIEW

**Comment by Limbicnation** [comment]
## Hermes Agent Code Review

**Verdict: Request Changes** — 2 bugs, 3 warnings, 3 suggestions. The node logic is clean but there are a dead parameter, a performance issue on large images, and a structlog import that will crash in standalone mode.

### Summary
- 2 files changed, +476/-3
- New file: `RemoveBackgroundAndResizeNode.py` (458 lines) — standalone PIL-only bg removal + resize
- Modified: `__init__.py` — registers the new node alongside existing ones
- Single commit: `36e20e6`

### Bugs (2)
1. **RemoveBackgroundAndResizeNode.py:164** — `edge_sensitivity` is declared in `remove_background_pil()` signature and documented, but **never used in the function body**. It is also **not passed** from `process()` (the call site only passes tolerance, foreground_bias, color_clusters, binary_threshold, dither_handling). This is a dead parameter — either wire it to `_dominant_bg()` (e.g., control the border strip width `bw` proportionally) or remove it.
2. **RemoveBackgroundAndResizeNode.py:23,30** — `structlog` is imported unconditionally and used for `log = structlog.get_logger()`. When running outside ComfyUI (standalone testing), `structlog` may not be installed, causing an immediate `ImportError`. Should use a try/except fallback to stdlib `logging`.

### Warnings (3)
1. **RemoveBackgroundAndResizeNode.py:116-138** — `_kmeans_colors()` runs on the **full pixel array** `arr.reshape(-1, C)`. For a 4K image that is 8.3M pixels × k clusters × max_iter, producing a `(8.3M, k)` distance matrix each iteration. This will OOM on large images. Should subsample to ~50K pixels before k-means (e.g., `np.random.choice(len(pixels), min(50_000, len(pixels)))`).
2. **RemoveBackgroundAndResizeNode.py:271** — `OUTPUT_NODE = False` — this node produces final output (IMAGE + MASK) that users will preview/save. Should be `OUTPUT_NODE = True`.
3. **RemoveBackgroundAndResizeNode.py:480** — `except Exception as e:` in `__init__.py` catches **all** exceptions (including `SyntaxError`, `NameError`), silently hiding real bugs in the new node. Should be `except ImportError as e:`.

### Suggestions (3)
1. **RemoveBackgroundAndResizeNode.py:23-29** — `ScalingMixin` is duplicated from `grabcut_nodes.py` (same class, same methods). Since both are in the same package, import from a shared `scaling.py` module instead of copy-pasting. This also applies to the different output_size presets — the new node has `[512x512, 768x768, 1024x1024, 1280x720, 1920x1080]` while grabcut_nodes uses `[512x512, 1024x1024, 2048x2048, custom]`. The power-of-8 sizes (64x64 through 2048x2048) are missing.
2. **RemoveBackgroundAndResizeNode.py:36-41** — `import comfy.utils` and `_COMFY_AVAILABLE` flag are imported but never used anywhere in the file. Remove dead code.
3. **RemoveBackgroundAndResizeNode.py:420** — The mask extraction `np.array(pil_rgba.split()[3])` is correct but `.split()` is a PIL method that returns a list of channel images. The alpha is at index 3 for RGBA. This works, but `.getchannel('A')` is more self-documenting and avoids the magic index.

### Looks Good
- Clean separation: background removal logic is pure PIL/numpy, no OpenCV dependency
- Good validation in `process()`: checks for None, empty tensor, wrong dimensions, wrong channel count
- `_dominant_bg()` smartly samples from borders + subsamples when too many pixels
- ComfyUI contract is correct: RETURN_TYPES, FUNCTION, CATEGORY, INPUT_TYPES structure
- RandomState with seed 42 for reproducibility
- `@torch.no_grad()` guard on the process method
- `__init__.py` cleanly merges all node mappings

---
*Reviewed by Hermes Agent*

File: RemoveBackgroundAndResizeNode.py [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+def remove_background_pil(
+    image: Image.Image,
+    tolerance: int = 30,
+    edge_sensitivity: float = 0.8,
+    color_clusters: int = 8,
+    foreground_bias: float = 0.7,
+    binary_threshold: int = 128,
+    dither_handling: bool = True,
+) -> Image.Image:
+    """
```
Comment (Limbicnation) [inline RemoveBackgroundAndResizeNode.py:119]: Bug: edge_sensitivity is declared in the function signature and documented in the docstring, but never used in the function body. It's also NOT passed from process() (line ~430). This is a dead parameter — either wire it to _dominant_bg() or remove it entirely.

File: RemoveBackgroundAndResizeNode.py [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+from __future__ import annotations
+
+import structlog
+from typing import Optional, Tuple
+
+import numpy as np
+import torch
+from PIL import Image
+
```
Comment (Limbicnation) [inline RemoveBackgroundAndResizeNode.py:23]: Bug: structlog is imported unconditionally. When running standalone (outside ComfyUI), structlog may not be installed, causing an immediate ImportError. Use a try/except fallback to stdlib logging.

File: RemoveBackgroundAndResizeNode.py [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+    Returns ``(n_clusters, channels)`` cluster centres.
+    Fallback when scikit-learn is unavailable.
+    """
+    pixels = arr.reshape(-1, arr.shape[-1]).astype(np.float32)
+    indices = np.random.RandomState(42).choice(
+        len(pixels), size=n_clusters, replace=False
+    )
+    centres = pixels[indices].copy()
+
+    for _ in range(max_iter):
```
Comment (Limbicnation) [inline RemoveBackgroundAndResizeNode.py:80]: Warning: _kmeans_colors() runs on the FULL pixel array. For a 4K image (8.3M pixels), this creates an (8.3M, n_clusters) distance matrix per iteration. This will OOM on large images. Add subsampling with _MAX_SAMPLES = 50_000.

File: RemoveBackgroundAndResizeNode.py [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+    RETURN_NAMES = ("image", "mask")
+    FUNCTION = "process"
+    CATEGORY = "image/processing"
+    OUTPUT_NODE = False
+
+    @classmethod
+    def INPUT_TYPES(cls):
+        return {
+            "required": {
+                "image": ("IMAGE", {"tooltip": "Input image tensor from upstream node"}),
```
Comment (Limbicnation) [inline RemoveBackgroundAndResizeNode.py:224]: Warning: OUTPUT_NODE = False — this node produces IMAGE + MASK output that users will preview or save. Should be OUTPUT_NODE = True.

File: __init__.py [hunk 5]
```diff
@@ -18,6 +18,21 @@
 # Import main transparency background remover nodes
 from .nodes import NODE_CLASS_MAPPINGS as NODES_MAPPINGS, NODE_DISPLAY_NAME_MAPPINGS as NODES_DISPLAY_MAPPINGS
 
+# Import standalone RemoveBackgroundAndResizeNode
+try:
+    from .RemoveBackgroundAndResizeNode import (
+        NODE_CLASS_MAPPINGS as REMBG_MAPPINGS,
+        NODE_DISPLAY_NAME_MAPPINGS as REMBG_DISPLAY_MAPPINGS,
+    )
```
Comment (Limbicnation) [inline __init__.py:26]: Warning: except Exception catches ALL exceptions including SyntaxError, NameError — silently hiding real bugs. Should be except ImportError.

File: RemoveBackgroundAndResizeNode.py [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+        "BICUBIC": Image.Resampling.BICUBIC,
+        "LANCZOS": Image.Resampling.LANCZOS,
+    }
+
+    @staticmethod
+    def _parse_size(size_str: str) -> Optional[Tuple[int, int]]:
+        """Parse ``WIDTHxHEIGHT`` string → ``(w, h)`` or ``None`` for ORIGINAL."""
+        if size_str == "ORIGINAL":
+            return None
+        try:
```
Comment (Limbicnation) [inline RemoveBackgroundAndResizeNode.py]: Suggestion: ScalingMixin is duplicated from grabcut_nodes.py. Extract to a shared scaling.py module to avoid drift.

File: RemoveBackgroundAndResizeNode.py [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+# ComfyUI import guards
+# ---------------------------------------------------------------------------
+try:
+    import comfy.utils
+
+    _COMFY_AVAILABLE = True
+except ImportError:
+    _COMFY_AVAILABLE = False
+    log.info("RemoveBackgroundAndResizeNode.comfyui_unavailable")
+
```
Comment (Limbicnation) [inline RemoveBackgroundAndResizeNode.py:29]: Suggestion: import comfy.utils and _COMFY_AVAILABLE are imported but never used. Remove the dead import guard.

**Review by Limbicnation — COMMENTED** [review]
Hermes Agent review - 2 bugs, 3 warnings, 2 suggestions. See inline comments.
